File: backend/app/schemas/shared_report.py
from pydantic import BaseModel, Field


class ShareReportRequest(BaseModel):
    user_id: int

    permission: str = Field(
        default="VIEW",
        pattern="^(VIEW|EXECUTE|EDIT)$",
    )


class UpdateSharePermissionRequest(BaseModel):
    permission: str = Field(
        pattern="^(VIEW|EXECUTE|EDIT)$",
    )


class SharedReportResponse(BaseModel):
    id: int
    report_id: int
    shared_with_user_id: int
    permission: str

    class Config:
        from_attributes = True

File: frontend/src/types/sharedReport.ts
export type SharePermission =
  | "VIEW"
  | "EXECUTE"
  | "EDIT";

export interface SharedReport {
  id: number;
  report_id: number;
  shared_with_user_id: number;
  permission: SharePermission;
}

export interface ShareReportPayload {
  user_id: number;
  permission: SharePermission;
}

export interface UpdateSharePermissionPayload {
  permission: SharePermission;
}

export interface ReportPermissionResponse {
  report_id: number;
  permission: SharePermission | null;
  has_access: boolean;
}